addons/pkg-npm/jaxrs/src/main/java/org/commonjava/indy/pkg/npm/jaxrs/NPMContentAccessHandler.java

```diff
@@ -0,0 +1,424 @@
+                {
+                    return;
+                }
+                path2 = Paths.get( idnode.asText(), "-", tarball ).toString();
+                content2 = anode.findPath( "data" ).asText();
+            }
+
+            if ( path1 == null || path2 == null )
+            {
+                return;
+            }
+            target1 = transfers.getCacheReference( new ConcreteResource( resource.getLocation(), path1 ) );
+            logger.info( "STORE {}", target1.getResource() );
+
+            target2 = transfers.getCacheReference( new ConcreteResource( resource.getLocation(), path2 ) );
+            logger.info( "STORE {}", target2.getResource() );
+        }
+        catch ( final IOException e )
+        {
+            logger.error( String.format( "[NPM] Json node parse failed for resource: %s. Reason: %s", resource,
+                                         e.getMessage() ), e );
+            response = formatResponse( e, builderModifier );
+
+            return;
+        }
+        try
+        {
+            if ( target1 == null || target2 == null )
+            {
+                return;
+            }
+            out1 = target1.openOutputStream( TransferOperation.UPLOAD, true, eventMetadata );
+            out1.write( content1.getBytes() );
+
+            out2 = target2.openOutputStream( TransferOperation.UPLOAD, true, eventMetadata );
+            out2.write( Base64.decodeBase64( content2 ) );
+
+            setGeneratedTransfers( transfer, target1, target2 );
+        }
+        catch ( final IOException e )
+        {
+            logger.error( String.format( "[NPM] Failed to store the generated targets: s% and s%. Reason: s%",
+                                         target1.getResource(), target2.getResource(), e.getMessage() ), e );
+            response = formatResponse( e, builderModifier );
+        }
+        finally
+        {
+            closeQuietly( out1 );
+            closeQuietly( out2 );
+        }
+    }
+
+    public void generateHttpMetadataHeaders( final Transfer transfer, final HttpServletRequest request,
+                                             final Response response )
+    {
+        if ( transfer == null || request == null || response == null )
+        {
+            return;
+        }
+
+        Response responseWithLastModified =
+                response.fromResponse( response ).lastModified( new Date( transfer.lastModified() ) ).build();
+
+        Transfer metaTxfr = transfer.getSiblingMeta( HttpExchangeMetadata.FILE_EXTENSION );
+        if ( metaTxfr == null )
+        {
+            if ( transfer.isDirectory() )
+            {
+                metaTxfr = transfer.getChild( HttpExchangeMetadata.FILE_EXTENSION );
+            }
+            else
+            {
+                return;
+            }
+        }
+
+        final HttpExchangeMetadata metadata = new StoreHttpExchangeMetadata( request, responseWithLastModified );
+        OutputStream out = null;
+        try
```

Comment:
Can you use try-with-resources when you do things like this? We still have some old code that uses this try/catch/finally approach for I/O, but it needs to be updated.

Instead, try-with-resources is cleaner:

```
try(OutputStream out = metaTxfr.openOutputStream(TransferOperation.GENERATE, false))
{
    if ( out != null )
    {
        [...]
    }
}
catch (IOException e)
{
    [...]
}
```